src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -2240,6 +2240,7 @@ public ObjectInstance(DefType type, AllocationSite allocationSite)
                 int size = type.InstanceByteCount.AsInt;
                 if (type.IsValueType)
                     size += type.Context.Target.PointerSize;
+                size = Math.Max(size, type.Context.Target.MinimumObjectSize - type.Context.Target.PointerSize);
```

Comment:
I think a better place to apply this padding would be in FrozenObjectNode.EncodeData.

FrozenStringNode.EncodeData is missing the padding too, but it is getting lucky - alignment of the next frozen object provides the padding.